### DISCUSSION TITLE
How to map gamepad axes

### DISCUSSION THREAD

**Discussion by Mitza21** [NicoHood/HID#355] [Q&A]
Hello, i `m just a beginer in coding, and i don`t understand how i suppose to map the ponentiometer for axes.
Thanks.

**Comment by dlbass** [comment]
I am new to this myself, so my answer is probably far from ideal, but it works as a starting point. Hope this helps someone else new to making custom controllers.

Since the potentiometer is an analog input that provides a range of values, as opposed to digital values of HIGH or LOW, you must capture the potentiometer's position using `analogRead()`. 

Once you have the current value of the potentiometer, it can simply be given to the Gamepad axis you want. Notice that your potentiometer probably gives values between 0 and 1024, and the documentation for the Gamepad X and Y axes range from -32768 to +32767, so you would need to figure out how you want to expand your 0-1024 to get full use of the available resolution. Or you can do what I did and just use Windows to calibrate with the values as they are.

Here's a sample of setting the Y-axis within `loop()`
```
  // Y-axis
  const int yRaw = analogRead(A0);     // read potentiometer at pin A0
  Gamepad.zAxis(yRaw);     // set position
  Gamepad.write();     // send value
  delay(100);     // debounce
```

**Reply by MyNameWasUnavailable** [reply]
I am trying to do something similar, I have an Arduino Uno flashd with Hoodloader2.  When I try and compile for the atmega16u2 on this device, I get the following:

Gamepad:30:31: error: 'A0' was not declared in this scope
   const int yRaw = analogRead(A0);     // read potentiometer at pin A0
                               ^~
C:\Users\xxxx\AppData\Local\Temp\arduino_modified_sketch_145379\Gamepad.ino:30:31: note: suggested alternative: 'B0'
   const int yRaw = analogRead(A0);     // read potentiometer at pin A0

If I change it to be something like the atmega32u4 it compiles correctly.

Am I missing something or are the analog pins not available for use when using a uno?

**Reply by MyNameWasUnavailable** [reply]
Ahh, NM, After reading the pin diagram for the hundreth time, I realised that the Pins I want to use, are in fact not connected.  oh well, back to the drawing board
